sklearn/mixture/_gaussian_mixture.py

```diff
@@ -726,15 +726,27 @@ def _initialize(self, X, resp):
                 covariances, self.covariance_type
             )
         elif self.covariance_type == "full":
+            n_features = self.precisions_init.shape[-1]
+            exchange_matrix = np.fliplr(np.eye(n_features))
             self.precisions_cholesky_ = np.array(
                 [
-                    linalg.cholesky(prec_init, lower=True)
+                    exchange_matrix
+                    @ linalg.cholesky(
+                        exchange_matrix @ prec_init @ exchange_matrix, lower=True
+                    )
+                    @ exchange_matrix
                     for prec_init in self.precisions_init
                 ]
             )
         elif self.covariance_type == "tied":
-            self.precisions_cholesky_ = linalg.cholesky(
-                self.precisions_init, lower=True
+            n_features = self.precisions_init.shape[-1]
+            exchange_matrix = np.fliplr(np.eye(n_features))
+            self.precisions_cholesky_ = (
+                exchange_matrix
+                @ linalg.cholesky(
+                    exchange_matrix @ self.precisions_init @ exchange_matrix, lower=True
+                )
+                @ exchange_matrix
```

Comment:
Would composing `np.fliplr` and `np.flipud` is equivalent, yet faster by orders or magnitude and more memory efficient (P and temporary results are not instantiated in this case):

```python
In [1]: import numpy as np

In [2]: n = 1000

In [3]: A = np.random.randn(n,n)

In [4]: %timeit np.flipud(np.fliplr(A))
1.05 µs ± 5.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [5]: P = np.fliplr(np.eye(n))

In [6]: %timeit P @ A @ P.T
2.32 s ± 58.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [7]: B = np.flipud(np.fliplr(A))

In [8]: C = P @ A @ P.T

In [9]: np.testing.assert_allclose(B, C)
```

Reply:
Yes, you are right. Using `np.flipud` and `np.fliplr` to permutate matrices is much faster. Changes have been made accordingly.

Reply:
Yes, they do not create arrays and simply return views.